src/Microsoft.Identity.Client/Features/ConfidentialClient/ConfidentialClientApplication.cs

```diff
@@ -389,6 +389,19 @@ async Task<AuthenticationResult> IConfidentialClientApplicationWithCertificate.A
             return await handler.CreateAuthorizationUriAsync().ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// This method should be used when you have a solution using ADAL 2.x and 
+        /// caching a refresh token, and you want to migrate to MSAL.NET.
+        /// During the migration process, it enables you to store in MSAL.NET token
+        ///  cache an access token and refresh token corresponding to <paramref="refreshToken"/>.
+        /// From there you will be able to use MSAL.NET new API, in particular
+        /// AcquireTokenSilentAsync() which will renew the user token.  
+        /// </summary>
+        async Task<AuthenticationResult> IByRefreshToken.AcquireTokenByRefreshTokenAsync(string refreshToken)
+        {
+            return await ExchangeRefreshTokenAsync(refreshToken).ConfigureAwait(false);
```

Comment:
You'll notice that all public methods in this class GuardMobileFrameworks() - yours should do too. The reason is that we hide the entire CCA at buildtime on mobile platforms, but we let it live at runtime so that we can throw decent exceptions.